pele/mindist/periodic_mindist.py

```diff
@@ -64,13 +57,18 @@ def __call__(self, coords1, coords2):
 
     def finalize_best_match(self, x1, best_x2):
         ''' do final processing of the best match '''
-        ca = CoordsAdapter(coords=best_x2)
-        ca.posRigid -= np.round(ca.posRigid / self.boxvec) * self.boxvec
-
-        dist = self.measure.get_dist(x1, best_x2)
+        x1 = x1.reshape(-1,3)
+        dx = x1 - best_x2
+        dx = np.round(dx / self.boxvec) * self.boxvec
+        self.transform.translate(best_x2, dx)
+        
+        # Calculate the periodic distance between the two structures
+        dx = x1 - best_x2
+        dx -= np.round(dx / self.boxvec) * self.boxvec
+        dist = np.linalg.norm(dx.flatten())
 #         if (dist - self.distbest) > 1e-6:
 #             raise RuntimeError(dist, self.distbest, "Permutational alignment has increased the distance metric")        
         if self.verbose:
 #         if True:
             print "finaldist", dist, "distmin", self.distbest
-        return dist, best_x2
+        return dist, best_x2.flatten()
```

Comment:
we should try to replace `flatten()` with `ravel()` throughout wherever appropriate.